lightning/src/onion_message/packet.rs

```diff
@@ -118,6 +118,13 @@ pub enum OnionMessageContents<T: CustomOnionMessageContents> {
 	Custom(T),
 }
 
+/// A type used within a variant of [`OnionMessageContents`].
+pub trait OnionMessageContentsType {}
```

Comment:
But we don't want a BOLT 12 handler to be able to send arbitrary ones. A custom handler can alway have their type wrap `OffersMessage` in a variant.

Reply:
Discussed offline and made the following changes:
- Renamed `OnionMessageContents` to `ParsedOnionMessageContents` and gave it `pub(super)` visibility
- Renamed `CustomOnionMessageContents` to `OnionMessageContents` and used it as a trait bound on most methods
- Changed Payload to hold `T` instead of `ParsedOnionMessageContents<T>` where `T: OnionMessageContents`
- Implemented `OnionMessageContents` for `OffersMessage` and `CustomOnionMessageContents`. The latter is needed so that arbitrary `Payload` contents can be read and specific contents can be written